File: RestaurantPOS.Android/app/src/main/java/com/example/restaurantpos/restaurantpo/smartorder/data/remote/dto/TableDtos.kt
package com.example.restaurantpos.restaurantpo.smartorder.data.remote.dto

import com.google.gson.annotations.SerializedName

data class TableDto(
    @SerializedName("id") val id: Int,
    @SerializedName("tableNumber") val tableNumber: String,
    @SerializedName("capacity") val capacity: Int,
    @SerializedName("isAvailable") val isAvailable: Boolean,
    @SerializedName("occupiedAt") val occupiedAt: String?,
    @SerializedName("floor") val floor: String,
    @SerializedName("isMerged") val isMerged: Boolean,
    @SerializedName("mergedGroupId") val mergedGroupId: Int?,
    @SerializedName("mergedTableNumbers") val mergedTableNumbers: String?
)

data class MergeTablesRequest(
    @SerializedName("tableIds") val tableIds: List<Int>
)

data class MergeTablesResponse(
    @SerializedName("groupId") val groupId: Int,
    @SerializedName("tableNumbers") val tableNumbers: String,
    @SerializedName("totalCapacity") val totalCapacity: Int,
    @SerializedName("tableCount") val tableCount: Int
)

File: RestaurantPOS.API/Models/TableDTOs.cs
namespace RestaurantPOS.API.Models;

public class MergeTablesRequest
{
    public List<int> TableIds { get; set; } = new();
}

public class MergeTablesResponse
{
    public int GroupId { get; set; }
    public string TableNumbers { get; set; } = string.Empty;
    public int TotalCapacity { get; set; }
    public int TableCount { get; set; }
}

public class MergedTableGroup
{
    public int GroupId { get; set; }
    public string? TableNumbers { get; set; }
    public int TotalCapacity { get; set; }
    public int TableCount { get; set; }
    public bool IsOccupied { get; set; }
}
